### PR TITLE
Added password prompt to Console

### PR DESCRIPTION
**Pull request by richardjh**
SImple prompt that doesn't echo back (or can echo *s). Useful for reading in passwords.
If accepted I'll write up the documentation too.

### CODE REVIEW

File: library/Zend/Console/Prompt/Password.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline library/Zend/Console/Prompt/Password.php]: 2015 ;)

File: tests/ZendTest/Console/Prompt/PasswordTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline tests/ZendTest/Console/Prompt/PasswordTest.php]: 2015 ;)

**Comment by richardjh** [comment]
Thanks @samsonasik I have updated copyright dates.

**Comment by Ocramius** [comment]
This is just stuck because the console prompt is waiting forever for input by the user (during the `while` loop execution). I'm changing the tests so that they mock out the console adapter.

**Comment by Ocramius** [comment]
Will need to rewrite some bits of this before pushing it out (due to security issues). Additionally, I think the `echo` functionality should just go away (security concerns).

**Comment by Ocramius** [comment]
This was merged into `develop` at zendframework/zf2@306c17a2818cbd87669fa71b1ae70a6e5bb96820

I had to rewrite some bits because the prompt was actually acting really weird when entering multiple rapidly subsequent chars: figured that (for security) the best solution was to clear the line at every keystroke instead.

**Pull request closed without merging**